language/move-binary-format/src/proptest_types/functions.rs

```diff
@@ -302,9 +302,13 @@ impl<'a> FnDefnMaterializeState<'a> {
         FunctionHandleIndex((self.function_handles.len() - 1) as TableIndex)
     }
 
-    fn get_signature_from_type_params(&mut self, abilities: &[AbilitySet]) -> Signature {
+    fn get_signature_from_type_params(
+        &mut self,
+        abilities: &[impl TypeParamGetConstraints],
```

Comment:
one thing you might want to do instead is `impl IntoIterator<Item = &AbilitySet>` that way you won't need this other trait